### PR TITLE
feat: Add Notes collection `fetchURL` method

### PR DESCRIPTION
**Pull request by taratatach**
This method calls the `/notes/:id/open` stack endpoint to fetch the
required details to build a url that points to the note in the Notes
application on the note's owner's Cozy.

This means that a shared note will return a NoteUrl that points to the
owner Cozy, with the original note's id and a share code to get
permissions to see the note.
If the note is not shared then the details will point to the client's
Cozy and no share code.

### CODE REVIEW

File: packages/cozy-stack-client/src/NotesCollection.js [hunk 1]
```diff
@@ -47,6 +53,26 @@ class NotesCollection extends DocumentCollection {
       data: { ...normalizeNote(resp.data), _deleted: true }
     }
   }
+
+  /**
+   * open - Returns the details to build the note's url
```
Comment (ptbrowne) [inline packages/cozy-stack-client/src/NotesCollection.js]:
```suggestion
   * Returns the details to build the note's url
```

File: packages/cozy-stack-client/src/NotesCollection.js [hunk 2]
```diff
@@ -47,6 +53,26 @@ class NotesCollection extends DocumentCollection {
...
+   *
+   * @see https://github.com/cozy/cozy-stack/blob/master/docs/notes.md#get-notesidopen
+   *
+   * @param {io.cozy.notes} note The note document to open
+   * @param {string}   note._id The note's id
+   *
+   * @returns {{ data }} The note's url details
+   */
+  async open({ _id }) {
```
Comment (ptbrowne) [inline packages/cozy-stack-client/src/NotesCollection.js]: When I read the description, I find that `fetchDetails` or `fetchURL` would be more appropriate ?

File: packages/cozy-stack-client/src/NotesCollection.js (hunk 1 above)
Comment (taratatach) [reply packages/cozy-stack-client/src/NotesCollection.js]: I'm just following the naming scheme of the comments in this file. Should I remove the name of the methods in all of them?

File: packages/cozy-stack-client/src/NotesCollection.js (hunk 2 above)
Comment (taratatach) [reply packages/cozy-stack-client/src/NotesCollection.js]: Sounds good to me.

File: packages/cozy-stack-client/src/NotesCollection.spec.js [hunk 3]
```diff
@@ -73,4 +73,58 @@ describe('NotesCollection', () => {
       expect(result.data._deleted).toBe(true)
     })
   })
+
+  describe('open', () => {
```
Comment (ptbrowne) [inline packages/cozy-stack-client/src/NotesCollection.spec.js]:
```suggestion
  describe('fetchURL', () => {
```

**Pull request merged**